### PR TITLE
add iolist_to_term

### PR DESCRIPTION
**Pull request by getong**

### CODE REVIEW

File: src/partisan_util.erl [hunk 1]
```diff
@@ -269,3 +270,29 @@ term_to_iolist_(T) ->
     %% fallback clause
     <<131, Rest/binary>> = term_to_binary(T),
     Rest.
+
+iolist_to_term([131, T]) ->
```
Comment (lrascao) [inline src/partisan_util.erl]: why the magic numbers?

Comment (getong) [reply src/partisan_util.erl]: term_to_iolist(Term) ->
     [131, term_to_iolist_(Term)].

**Comment by cmeiklejohn** [comment]
@Vagabond, can you approve this one?

**Comment by Vagabond** [comment]
What is the purpose of this function?

**Comment by getong** [comment]
Because the 'decode/1` function only use `binary_to_term/1`, only the `default_encode` match this. So all tcp data just use these two functions.

**Comment by cmeiklejohn** [comment]
Do not use default_encode for the TCP messages -- you'll lose out on the benefits of binary sharing.  Instead of using encode for the on disk data, use default_encode and then decode it with binary_to_term.

**Comment by getong** [comment]
@cmeiklejohn  I think this patch is just as you described.

**Comment by cmeiklejohn** [comment]
No, this patch is still wrong.

You're sending messages in this patch with ``default_encode``.  That is the *wrong* encoding to use for wire messages: you need to use ``encode``.

**Comment by cmeiklejohn** [comment]
``encode`` is optimized for the wire; if you want to store to disk, use ``default_encode``.

**Comment by getong** [comment]
ok, I will use encode, and write a decode function, but need time.

**Pull request closed without merging**